composer/loggers/cometml_logger.py

```diff
@@ -0,0 +1,102 @@
+            under Uncategorized Experiments.
+        log_code (bool): Whether to log your code in your experiment (default: ``False``).
+        log_graph (bool): Whether to log your computational graph in your experiment
+            (default: ``False``).
+        name (str, optional): The name of your experiment. If not specified, it will be set
+            to :attr:`.State.run_name`.
+        rank_zero_only (bool, optional): Whether to log only on the rank-zero process.
+            (default: ``False``).
+        exp_kwargs (Dict[str, Any], optional): Any additional kwargs to
+            comet_ml.Experiment(see
+            `Comet documentation <https://www.comet.com/docs/v2/api-and-sdk/python-sdk/reference/Experiment/>`_).
+    """
+
+    def __init__(
+        self,
+        workspace: Optional[str] = None,
+        project_name: Optional[str] = None,
+        log_code: bool = False,
+        log_graph: bool = False,
+        name: Optional[str] = None,
+        rank_zero_only: bool = True,
+        exp_kwargs: Optional[Dict[str, Any]] = None,
+    ) -> None:
+        try:
+            from comet_ml import Experiment
+        except ImportError as e:
+            raise MissingConditionalImportError(extra_deps_group='comet_ml',
+                                                conda_package='comet_ml',
+                                                conda_channel='conda-forge') from e
+
+        self._enabled = (not rank_zero_only) or dist.get_global_rank() == 0
+
+        if exp_kwargs is None:
+            exp_kwargs = {}
+
+        if workspace is not None:
+            exp_kwargs['workspace'] = workspace
+
+        if project_name is not None:
+            exp_kwargs['project_name'] = project_name
+
+        exp_kwargs['log_code'] = log_code
+        exp_kwargs['log_graph'] = log_graph
+
+        self.name = name
+        self._rank_zero_only = rank_zero_only
+        self._exp_kwargs = exp_kwargs
+        self.experiment: Optional[Experiment] = None
+
+    def init(self, state: State, logger: Logger) -> None:
+        import comet_ml
+        del logger  # unused
+
+        # Use the logger run name if the name is not set.
+        if self.name is None:
+            self.name = state.run_name
+
+        # Adjust name and group based on `rank_zero_only`.
+        if not self._rank_zero_only:
+            self.name += f'-rank{dist.get_global_rank()}'
+
+        if self._enabled:
+            self.experiment = comet_ml.Experiment(**self._exp_kwargs)
+            self.experiment.set_name(self.name)
+
+    def log_metrics(self, metrics: Dict[str, Any], step: Optional[int] = None) -> None:
+        if self._enabled:
+            assert self.experiment is not None
+            self.experiment.log_metrics(dic=metrics, step=step)
+
+    def log_hyperparameters(self, hyperparameters: Dict[str, Any]):
+        if self._enabled:
+            assert self.experiment is not None
```

Comment:
Same comment as in `log_metrics()` method.

Reply:
fixed